### PR TITLE
feat: Add caching support

### PR DESCRIPTION
**Pull request by s0nicboOm**
Feature Addition: Add support for Caching

### CODE REVIEW

File: pyproject.toml
Comment (ab93) [inline pyproject.toml:1]: Can you update the project version here also?

File: numaprom/udf/threshold.py [hunk 1]
```diff
@@ -60,7 +60,8 @@ def threshold(_: List[str], datum: Datum) -> list[tuple[str, bytes]]:
         ]
 
     # load threshold artifact
-    model_registry = RedisRegistry(client=get_redis_client_from_conf())
+    local_cache = LocalLRUCache(ttl=3600)  # setting ttl to 1 hour
```
Comment (ab93) [inline numaprom/udf/threshold.py:63]: Let's use a default constant

Comment (ab93) [reply numaprom/udf/threshold.py:63]: Or get from env

**Review by ab93 — COMMENTED** [review]
Minor comments

File: numaprom/udf/threshold.py (hunk 1 above)
Comment (s0nicboOm) [reply numaprom/udf/threshold.py]: Fair point.

File: pyproject.toml
Comment (s0nicboOm) [reply pyproject.toml:1]: done patch version

**Review by ab93 — APPROVED** [review]

File: pyproject.toml [hunk 2]
```diff
@@ -21,7 +21,7 @@ repository = "https://github.com/numaproj/numalogic-prometheus"
 python = "~3.10"
 redis = {extras = ["hiredis"], version = "^4.5" }
 pynumaflow = "~0.4.1"
-numalogic = {version = "0.4a0", extras = ["redis"], allow-prereleases = true}
+numalogic = {version = "v0.4.0", extras = ["redis"], allow-prereleases = true}
```
Comment (ab93) [inline pyproject.toml:24]:
```suggestion
numalogic = { version = "~0.4", extras = ["redis"] }
```

**Pull request merged**